internal_packages/keybase/lib/encrypt-button.cjsx

```diff
@@ -0,0 +1,140 @@
+{Utils, DraftStore, React} = require 'nylas-exports'
+PGPKeyStore = require './pgp-key-store'
+{RetinaImg} = require 'nylas-component-kit'
+pgp = require 'kbpgp'
+_ = require 'underscore'
+
+class EncryptMessageButton extends React.Component
+
+  @displayName: 'EncryptMessageButton'
+
+  # require that we have a draft object available
+  @propTypes:
+    draftClientId: React.PropTypes.string.isRequired
+
+  constructor: (props) ->
+    super(props)
+    @state = @_getStateFromStores()
+
+    # maintain the state of the button's toggle
+    @state.currentlyEncrypted = false
+
+    # store the message's plaintext in case the user wants to edit further after
+    # hitting the "encrypt" button (i.e. so we can "undo" the encryption)
+    @state.plaintext = ""
+
+    # store the message's body here, for comparison purposes (so that if the
+    # user edits an encrypted message, we can revert it)
+    @state.cryptotext = ""
+
+  componentDidMount: ->
+    @unlistenKeystore = PGPKeyStore.listen(@_onKeystoreChange, @)
+    # periodically check to see if we should get any keys
+    @interval = setInterval(@_onTrigger, 100)
```

Comment:
We actually changed this earlier today so that your `EncryptMessageButton` will receive a "draft" prop, and you can implement `componentWillReceiveProps` to see if there are any new participants. Ping me and we can try implementing it.